packages/fx-core/src/component/resource/botService/botService.ts

```diff
@@ -86,15 +79,47 @@ export class BotService extends AzureResource {
     await actionContext?.progressBar?.next(ProgressMessages.provisionBot);
     // init bot state
     context.envInfo.state[ComponentNames.TeamsBot] ||= {};
-    const aadRes = await createBotAAD(context);
-    if (aadRes.isErr()) return err(aadRes.error);
+    const teamsBotState = context.envInfo.state[ComponentNames.TeamsBot];
+
+    let botRegistration: BotRegistration | undefined = undefined;
     if (context.envInfo.envName === "local") {
-      const botConfig = aadRes.value;
-      const regRes = await createBotRegInAppStudio(botConfig, context);
-      if (regRes.isErr()) return err(regRes.error);
+      botRegistration = new LocalBotRegistration();
+    } else {
+      botRegistration = new RemoteBotRegistration();
     }
+
+    const solutionConfig = context.envInfo.state.solution as v3.AzureSolutionConfig;
+    const resourceNameSuffix = solutionConfig.resourceNameSuffix
+      ? solutionConfig.resourceNameSuffix
+      : uuid.v4();
+    const aadDisplayName = ResourceNameFactory.createCommonName(
+      resourceNameSuffix,
+      context.projectSetting.appName,
+      MaxLengths.AAD_DISPLAY_NAME
+    );
+
+    const botConfig: IBotAadCredentials =
+      context.envInfo.config.bot?.appId && context.envInfo.config.bot?.appPassword
+        ? {
+            botId: context.envInfo.config.bot?.appId,
+            botPassword: context.envInfo.config.bot?.appPassword,
+          }
+        : {
+            botId: teamsBotState.botId,
+            botPassword: teamsBotState.botPassword,
+          };
+
+    const regRes = await botRegistration.createBotRegistration(
```

Comment:
When provision remote, it throws NoImplementError?

Reply:
Does this mean we don't create Azure bot service by bicep?

Reply:
No, because `RemoteBotRegistration` have empty implementations which will do nothing.

Reply:
> Does this mean we don't create Azure bot service by bicep?

Yeah, it's handled by arm/bicep so we just create a bot AAD app for remote and leave Azure bot service provision and configurations to arm/bicep.

Reply:
I read it wrong. RemoteBotRegistration.createBotRegistration call super.createBotAadApp. I thought it calls super.createBotRegistration, which is not implemented.